opengrok-tools/src/main/python/opengrok_tools/utils/commandsequence.py

```diff
@@ -173,13 +176,17 @@ def run(self):
 
     def run_cleanup(self):
         """
-        Call cleanup in case the sequence failed or termination was requested.
+        Call cleanup sequence in case the command sequence failed
+        or termination was requested.
         """
-        if self.cleanup:
-            if is_web_uri(self.cleanup.get("command")[0]):
-                self.call_rest_api(self.cleanup)
+        if self.cleanup is None:
+            return
+
+        for cleanup_cmd in self.cleanup:
+            if is_web_uri(cleanup_cmd.get("command")[0]):
+                self.call_rest_api(cleanup_cmd)
             else:
-                command_args = self.cleanup.get("command")
+                command_args = cleanup_cmd.get("command")
                 self.logger.debug("Running cleanup command '{}'".
                                   format(command_args))
                 cmd = Command(command_args,
```

Comment:
it looks like it works but just to verify - when one of the commands fails, do the following commands get executed?

Reply:
they should, let me augment the test case.